### ISSUE TITLE
Playpen panics on valid code

### ISSUE THREAD

**Issue by klingtnet** [rust-lang/rust-playpen#178]
I've written an example that demonstrates how to hand-back ownership but the [playpen](http://is.gd/iM00NQ)) panics when I try to run it:

```
thread '<main>' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore/option.rs:367
playpen: application terminated with error code 101
```

The following snippet runs fine when I compile it using rustc stable `1.6.0`, beta or nightly:

```rust
fn do_something(v: Vec<i32>) -> Vec<i32> {
    // do something
    // hand back ownership
    v
}

fn main() {
    let v = vec![1,2,3];
    let v = do_something(v);
    // handing back ownership would get very tedious
    println!("{}", v[0]);
}
```

**Comment by klingtnet** [comment]
A more simple example that panics in the playpen as well:

```rust
fn main() {
    let x = 1;
    println!("{}", x);
    println!("{}", x);
}
```

Prints `1\n1` as expected when compiled with rustc.

**Comment by klingtnet** [comment]
UPDATE: Regardless of the code to run, the playpen always responsds with: `thread '<main>' panicked at 'called `Option::unwrap()` on a `None` value'...`

**Comment by rigdent** [comment]
+1 on the last comment.

**Comment by jonas-schievink** [comment]
It now always fails with
```
error: linking with `cc` failed: exit code: 1
...
note: collect2: fatal error: ld terminated with signal 31 [Bad system call]
```

Maybe a system update installed a new version of gcc/ld which uses a new system call not in the whitelist? In that case, the fix should be easy, check which syscalls are needed and add them to the whitelist (probably using playpen's learning mode).

**Comment by jonas-schievink** [comment]
Looks binutils (which contains ld) was updated yesterday: https://www.archlinux.org/packages/core/x86_64/binutils/

**Comment by alexcrichton** [comment]
Fixed now, thanks for the report!

**Comment by dwrensha** [comment]
I am curious about what caused the problem. @alexcrichton can you share any details?

**Comment by alexcrichton** [comment]
I think that @jonas-schievink's diagnosis was spot on in that the binutils package in Arch updated, so we started using a new linker. This must have in turn required some new syscall (hence the termination with a bad system call), and the problem was learning what syscall was needed. Unfortunately the version of playpen that we're using on the AMI in prod is too old to be able to learn about syscalls from child processes, so I had to install an Arch VM locally to install an updated Arch to install an updated playpen to learn about which syscall was needed. Turns out it was just the `sysinfo` syscall!

Unfortunately I'm not actually sure what the `Option::unwrap()` on `None` came from. I wasn't able to reproduce that error exactly anywhere outside of the web interface itself. It may be related to the libsyntax/libterm changes recently in the compiler, but I'm not sure :(
